src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -745,4 +745,30 @@ impl CatalogState {
             diff,
         }
     }
+
+    pub fn pack_storage_usage_update(
+        &self,
+        storage_object_id: &str,
+        object_size_bytes: u64,
+        timestamp: DateTime<Utc>,
+    ) -> BuiltinTableUpdate {
+        let table = self.resolve_builtin_table(&MZ_STORAGE_USAGE);
+        let row = Row::pack_slice(&[
+            Datum::String(storage_object_id),
+            Datum::Int64(
+                object_size_bytes
+                    .try_into()
+                    // Unless one object has over 9k petabytes of storage...
+                    .expect("Storage bytes size should not overflow i64"),
+            ),
+            Datum::TimestampTz(timestamp),
+        ]);
+        // TODO: what should this be? Table is append-only
```

Comment:
Heads up that the data in system tables gets deleted on restart. @mjibson can get you set up with what he did for the audit log, which knows how to keep data around to survive a restart.

Reply:
@benesch - Are you suggesting that storage usage data would make more sense as part of the audit log, or just that I could use a similar strategy to persist the data across restarts?

Reply:
> just that I could use a similar strategy to persist the data across restarts?

This, yes! I think it's probably easiest to talk about what I'm not suggesting—I'm _not_ suggesting that we merge this table with the `mz_audit_events` table.

But, definitely you should persist the contents of this table in the same manner as `mz_audit_events` its contents. Ideally you'd sharing as much code as possible—and it might be that we describe that as "recording storage usage data in the audit log", even though it gets presented in a different table than `mz_audit_events`.

Reply:
After gaining a better understanding of how the audit log code is working, I didn't see a large benefit to directly reusing the audit log library for the storage usage table. The schemas and tolerances are different enough that it would have required changes to the overall package that didn't seem helpful for its main purpose, and would couple together two processes that don't necessarily need to be compatible. If we think there is a third use for this functionality, it might be worth considering.
That said, if it seems like there are extra maintenance burdens due to the duplicated code pathways I'd be happy to chat more about a better design.